src/derive/curve.rs

```diff
@@ -564,7 +564,11 @@ macro_rules! new_curve_impl {
                 let z_inv = z.invert().unwrap_or($base::zero());
                 let p_x = x * z_inv;
                 let p_y = y * z_inv.square();
-                let p = $name { x:p_x, y:p_y, z };
+                let p = $name {
+                    x:p_x,
+                    y:$base::conditional_select(&p_y, &$base::one(), z.is_zero()),
+                    z
+                };
```

Comment:
Something was added here that disrupted the ASM tests. You should double check first! See the latest test reports.